kubernetes/iceberg/deploy-spark.sh

```diff
@@ -0,0 +1,244 @@
+    exit 1
+fi
+
+# =============================================================================
+# RESOLVE MINIO POD IP
+# =============================================================================
+# Get the actual IP address of the MinIO pod for hostAliases configuration
+# This IP will be used to resolve bucket-style S3 URLs
+
+MINIO_IP=$(kubectl get pod -n "$NAMESPACE" -l "$MINIO_POD_SELECTOR" -o jsonpath='{.items[0].status.podIP}' 2>/dev/null)
+
+if [[ -z "$MINIO_IP" ]]; then
+    echo "ERROR: Failed to resolve MinIO pod IP"
+    echo "This could indicate:"
+    echo "  - MinIO pod is not running"
+    echo "  - Network policies blocking access"
+    echo "  - Insufficient RBAC permissions"
+    echo ""
+    echo "Debug steps:"
+    echo "  kubectl get pods -n $NAMESPACE -l $MINIO_POD_SELECTOR"
+    echo "  kubectl describe pod -n $NAMESPACE -l $MINIO_POD_SELECTOR"
+    exit 1
+fi
+
+echo "✓ MinIO IP resolved: $MINIO_IP"
+
+# =============================================================================
+# GENERATE SPARK DEPLOYMENT CONFIGURATION
+# =============================================================================
+# Create a deployment configuration with:
+# 1. Dynamic IP injection for hostAliases
+# 2. Proper resource allocation for Spark workloads
+# 3. Health checks for reliable operation
+# 4. Environment variables for S3 authentication
+
+cat > "$SPARK_OUTPUT" << EOF
+# =============================================================================
+# SPARK DEPLOYMENT FOR ICEBERG ANALYTICS
+# =============================================================================
+# Generated by deploy-spark.sh with dynamic MinIO IP resolution
+# This deployment provides a Jupyter notebook environment with Spark and Iceberg
+# for querying Redpanda topics as analytical tables
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: spark-iceberg
+  namespace: iceberg-lab
+  labels:
+    app: spark-iceberg
+    deployment-method: dynamic-script
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: spark-iceberg
+  template:
+    metadata:
+      labels:
+        app: spark-iceberg
+      annotations:
+        # Track deployment revisions for rollback capability
+        deployment.kubernetes.io/revision: "$(date +%s)"
+        # Store MinIO IP for debugging and monitoring
+        minio.ip: "$MINIO_IP"
+    spec:
+      # =============================================================================
+      # DNS RESOLUTION FOR S3 BUCKET-STYLE URLS
+      # =============================================================================
+      # Iceberg and Spark sometimes generate S3 URLs in bucket-style format
+      # (bucket.hostname) which don't resolve in Kubernetes DNS
+      # These hostAliases map bucket-style names to the MinIO pod IP
+      hostAliases:
+        - ip: "$MINIO_IP"  # MinIO pod IP (resolved dynamically)
```

Comment:
I feel like this would be easier to achieve with either a coreDNS modification an init container that does some /etc/host hackery in the spark container.